prqlc/prql-compiler/tests/integration/sql.rs

```diff
@@ -166,13 +177,34 @@ fn test_stdlib_string() {
       CHAR_LENGTH(name) AS name_length,
       SUBSTRING(name, 3, 5) AS name_substring,
       REPLACE(name, 'pika', 'chu') AS name_replace,
-      (name) LIKE CONCAT(('pika'), '%') AS name_starts_with,
-      (name) LIKE CONCAT('%', ('pika'), '%') AS name_contains,
-      (name) LIKE CONCAT('%', ('pika')) AS name_ends_with
+      name LIKE CONCAT('pika', '%') AS name_starts_with,
+      name LIKE CONCAT('%', 'pika', '%') AS name_contains,
+      name LIKE CONCAT('%', 'pika') AS name_ends_with
+    FROM
+      employees
+    "#
+    );
+}
+
+#[rstest]
+#[case::generic(sql::Dialect::Generic, "LIKE CONCAT('%', 'pika', '%')")]
+#[case::sqlite(sql::Dialect::SQLite, "LIKE '%' || 'pika' || '%'")] // `CONCAT` is not supported in SQLite
```

Comment:
Interesting. I had heard of `rstest` but haven't used it before. (I use pytest a lot, which I think is similar)

Happy to try this! (Maybe @aljazerzen has thoughts?)

(alternatively, we could have a normal loop and supply different names for the snapshot name)

Reply:
I'm a huge fan of pytest and the parametrize + fixtures mechanisms.
rstest is quite similar and I feel like it would be a great value to cover many dialects since prql will probably need more tests to ensure compatibility across all the supported DBs. In my company we support 5 SQL databases/DWs and everything is parametrized on that.
But I can definitely remove that if you prefer

Reply:
Very happy to try it for a while at least!

Reply:
I'm +1 for rstest - on the macro-expansion level it does basically the same thing as `for_each_file!`, but offers much more flexibility.